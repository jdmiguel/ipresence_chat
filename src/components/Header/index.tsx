import React from 'react';

interface IHeaderProps {
  onClick: () => void;
  unreadMessagesCounter: number;
  isTyping: boolean;
}

const Header: React.FC<IHeaderProps> = ({
  onClick,
  unreadMessagesCounter,
  isTyping,
}) => (
  <header>
    <button type="button" onClick={onClick}></button>
    <div className="header-profile">
      <i className="material-icons">account_circle</i>
      <div>
        <h1>Ipresence User</h1>
        <h2 className={isTyping ? 'typing' : ''}>
          {isTyping ? 'Typing...' : 'Online'}
        </h2>
      </div>
    </div>
    <div>
      <h3>{`${unreadMessagesCounter} new messages`}</h3>
    </div>
  </header>
);

export default Header;
