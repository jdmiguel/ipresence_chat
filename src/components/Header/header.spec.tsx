import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Header from '.';

const HEADER_PROPS = {
  onClick: () => {},
  unreadMessagesCounter: 10,
  isTyping: false
};

describe('Component: Header', () => {
  it('should render and display Online text', () => {
    const { container } = render(
      <Header {...HEADER_PROPS} />
    );
    const header = container.firstChild as HTMLDivElement;
    const stateText = header.querySelector('h2')?.innerHTML;

    expect(header).toMatchSnapshot();
    expect(stateText).toBe('Online');
  });

  it('should display Typing text', () => {
    const { container } = render(
      <Header {...HEADER_PROPS} isTyping={true} />
    );
    const header = container.firstChild as HTMLDivElement;
    const stateHeading = header.querySelector('h2');
    const stateText = stateHeading?.innerHTML;

    expect(stateHeading).toHaveClass('typing');
    expect(stateText).toBe('Typing...');
  });

  it('should call function onClick', () => {
    const handleClick = jest.fn();
    const { container } = render(
      <Header {...HEADER_PROPS} onClick={handleClick}/>
    );
    const header = container.firstChild as HTMLDivElement;
    const button = header.querySelector('button');

    fireEvent.click(button as HTMLButtonElement);
    expect(handleClick).toHaveBeenCalled();
  });
});