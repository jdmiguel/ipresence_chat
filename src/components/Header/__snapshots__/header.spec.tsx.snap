// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Header should render and display Online text 1`] = `
<header>
  <button
    type="button"
  />
  <div
    class="header-profile"
  >
    <i
      class="material-icons"
    >
      account_circle
    </i>
    <div>
      <h1>
        Ipresence User
      </h1>
      <h2
        class=""
      >
        Online
      </h2>
    </div>
  </div>
  <div>
    <h3>
      10 new messages
    </h3>
  </div>
</header>
`;
